spec/support/view_component/matchers.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Support
+  module ViewComponent
+    module Matchers
+      extend RSpec::Matchers::DSL
+
+      define :have_rendered_component do |component_class|
+        match do |rendered|
+          @stubs = _test_context.components.select { |stubbed| stubbed.instance.is_a? component_class }
+          @matching = @stubs.select { |stubbed| arguments.args_match?(*stubbed.args) }
+          @matching.any? do |stubbed|
+            rendered.content.include?(stubbed.output)
+          end
+        end
+
+        chain :with do |*args|
+          @arguments = RSpec::Mocks::ArgumentListMatcher.new(*args)
+        end
+
+      private
+
+        def arguments
+          @arguments ||= RSpec::Mocks::ArgumentListMatcher.new(any_args)
+        end
+      end
```

Comment:
Can we move this into the `match`? I find a `private` method for a `define` scope to be a bit of a mind bender.

Reply:
Yeah it is a bit odd. Block given to `define` is evaluated in the context of the new matcher class, so in fact `define` is very similar to `class` (but with a symbol rather than constant) which makes it slightly easier to read. We could move this, however this pattern will keep appearing from time to time - see spec/support/delayed_job.rb`